admiral/component_envs/attacking.py

```diff
@@ -0,0 +1,83 @@
+
+from admiral.envs import Agent
+from admiral.component_envs.position import GridPositionAgent
+from admiral.component_envs.team import TeamAgent
+from admiral.component_envs.component import Component
+from admiral.component_envs.death_life import DyingAgent
+
+class GridAttackingAgent(Agent):
+    def __init__(self, attack_range=None, attack_strength=None, **kwargs):
+        assert attack_range is not None, "attack_range must be a nonnegative integer"
+        self.attack_range = attack_range
+        assert attack_strength is not None, "attack_strength must be a nonnegative number"
+        self.attack_strength = attack_strength
+        super().__init__(**kwargs)
+    
+    @property
+    def configured(self):
+        """
+        Determine if the agent has been successfully configured.
+        """
+        return super().configured and self.attack_range is not None and self.attack_strength is not None
+
+class GridAttackingComponent(Component):
+    def __init__(self, agents=None, **kwargs):
+        assert type(agents) is dict, "agents must be a dict"
+        for agent in agents.values():
+            assert isinstance(agent, GridPositionAgent)
```

Comment:
Add comment: attacks are based on the agents' position, so the agents must be grid positional agents